lib/view/home_page.dart

```diff
@@ -0,0 +1,1849 @@
+import 'dart:io';
+
+import 'package:dpip/util/extension.dart';
+import 'package:dpip/view/report.dart';
+import 'package:dpip/view/setting/location.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:intl/intl.dart';
+import 'package:timezone/timezone.dart';
+
+import '../core/utils.dart';
+import '../global.dart';
+import '../model/partial_earthquake_report.dart';
+import '../util/dist_code.dart';
+import '../util/intensity_color.dart';
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePage();
+}
+
+class Areas {
+  static List<String> getOptions(currentArea) {
+    return [currentArea]; // 鄉鎮列表
+  }
+}
+
+class Cal {
+  double percentToPixel(double percent, BuildContext context) {
+    double screenWidth = MediaQuery.of(context).size.width;
+    return percent / 100 * screenWidth;
+  }
+}
+
+Future<int?> getZipCodeForArea(String area) async {
+  final data = await DistCodeUtil.readJsonFile();
+  return DistCodeUtil.getZipCode(area, data);
+}
+// class IntColor {
+//   static const Map<int, Color> _colors = {
+//     0: Color(0xff202020),
+//     1: Color(0xff003264),
+//     2: Color(0xff0064c8),
+//     3: Color(0xff1e9632),
+//     4: Color(0xffffc800),
+//     5: Color(0xffff9600),
+//     6: Color(0xffff6400),
+//     7: Color(0xffff0000),
+//     8: Color(0xffc00000),
+//     9: Color(0xff9600c8),
+//   };
+//
+//   Color intColor(int intensity) {
+//     return _colors[intensity] ?? Color(0xFF202020);
+//   }
+// }
+
+class TempColor {
+  List<Color> tempColors = [
+    const Color(0xFF006060),
+    const Color(0xFF00AFAF),
+    const Color(0xFF00FFFF),
+    const Color(0xFF3AAA50),
+    const Color(0xFFFFFF00),
+    const Color(0xFFFF8A00),
+    const Color(0xFFFF0000),
+    const Color(0xFFFF00CA),
+    const Color(0xFF6040B0),
+  ];
+
+  Color getColorForTemp(double temp) {
```

Comment:
```suggestion
  Color getTemperatureColor(double temperature) {
```
Temp 通常會指暫存 temporary 的意思
溫度直接把 Temperature 寫出來會比較好